src/test/inputfetcher_tests.cpp

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <coins.h>
+#include <common/system.h>
+#include <inputfetcher.h>
+#include <primitives/block.h>
+#include <primitives/transaction.h>
+#include <test/util/random.h>
+#include <test/util/setup_common.h>
+#include <uint256.h>
+#include <util/transaction_identifier.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <cstdint>
+#include <stdexcept>
+#include <string>
+#include <unordered_set>
+
+BOOST_AUTO_TEST_SUITE(inputfetcher_tests)
+
+struct InputFetcherTest : BasicTestingSetup {
+    CBlock CreateBlock(int32_t num_txs)
+    {
+        CBlock block;
+        CMutableTransaction coinbase;
+        coinbase.vin.emplace_back();
+        block.vtx.push_back(MakeTransactionRef(coinbase));
+
+        Txid prevhash{Txid::FromUint256(uint256(1))};
+
+        for (auto i{1}; i < num_txs; ++i) {
+            CMutableTransaction tx;
+            const auto txid{m_rng.randbool() ? Txid::FromUint256(uint256(i)) : prevhash};
+            tx.vin.emplace_back(COutPoint(txid, 0));
+            prevhash = tx.GetHash();
+            block.vtx.push_back(MakeTransactionRef(tx));
+        }
+
+        return block;
+    }
+};
+
+BOOST_FIXTURE_TEST_CASE(fetch_inputs, InputFetcherTest)
+{
+    SeedRandomForTest(SeedRand::ZEROS);
+
+    const auto cores{GetNumCores()};
+    const auto num_txs{m_rng.randrange(cores * 10)};
+    const auto block{CreateBlock(num_txs)};
+    const auto batch_size{m_rng.randrange<int32_t>(block.vtx.size() * 2)};
+    const auto worker_threads{m_rng.randrange(cores * 2)};
+    InputFetcher fetcher{batch_size, worker_threads};
```

Comment:
In 3c201bcffc1d7e382e8afa9a88750a4c261c1cf8 "tests: add inputfetcher tests"
You can set this up in `InputFetcherTest` so that you don't have to repeat it in the rest of the tests.

Reply:
Done.